src/components/Datatable/Datatable.vue

```diff
@@ -0,0 +1,299 @@
+            <Button @click="sortColumns(columns[0])">Sort: A-Z</Button></slot
+          >
+        </th>
+      </tr>
+
+      <tr class="table-row">
+        <th
+          v-for="(column, columnIndex) in columns"
+          class="table-header"
+          :style="{ textAlign: column.align ? column.align : 'start' }"
+          :key="columnIndex"
+          @click="sortColumns(column)"
+        >
+          <span class="icon-container">
+            <slot name="column" :column="column">
+              <slot
+                :name="'column-' + columns[columnIndex].value"
+                :column="column"
+              >
+                {{ column.text }}
+              </slot>
+
+              <i
+                v-if="currentSortDir == 'asc'"
+                class="pi pi-arrow-up-line pi-1x"
+              />
+              <i
+                v-else-if="currentSortDir == 'desc'"
+                class="pi pi-arrow-down-line pi-1x"
+              />
+
+              <div v-else class="rotate-arrow">
+                <i class="pi pi-code-line pi-1x" />
+              </div>
+            </slot>
+          </span>
+        </th>
+      </tr>
+      <tr class="table-row">
+        <td :colspan="columns.length" class="search-input">
+          <Input v-model="search" placeholder="Search...">
+            <template v-slot:prepend>
+              <div class="search-icon">
+                <i class="pi pi-search-line pi-1x" />
+              </div>
+            </template>
+          </Input>
+        </td>
+      </tr>
+    </thead>
+
+    <tbody class="table-body">
+      <tr v-for="(row, rowIndex) in sortedColumns" :key="rowIndex">
+        <td
+          v-for="column in columns"
+          :key="column.value"
+          :align="column.align || 'start'"
+          class="table-cell"
+        >
+          <slot name="item" :item="row">
+            <slot :name="`item-${column.value}`" :item="row">
+              {{ row[column.value] }}
+            </slot>
+          </slot>
+        </td>
+      </tr>
+
+      <tr>
+        <td
+          v-if="sortedColumns.length === 0"
+          :colspan="columns.length"
+          align="center"
+          class="no-search-results"
+        >
+          <div>No results found</div>
+          <Button @click="clearSearch" miter width="170px" height="36px"
+            >Clear search</Button
+          >
+        </td>
+      </tr>
+    </tbody>
+  </table>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from 'vue'
+import Button from '../Button/Button.vue'
+import Input from '../Input/Input.vue'
+
+interface Columns {
+  text: string
+  value: string
+  align: string
+  sort: void
+}
+
+export default defineComponent({
+  name: 'DataTable',
+  components: { Button, Input },
+  props: {
+    columns: {
+      type: Array as PropType<Columns[]>,
+      required: true,
+      default: function () {
+        return []
+      }
```

Comment:
You can still use an arrow function like you had before

```suggestion
      default: () => []
```